src/py21cmmc/mcmc.py

```diff
@@ -172,19 +204,56 @@ def run_mcmc(
     if log_level_21CMMC is not None:
         logging.getLogger("21CMMC").setLevel(log_level_21CMMC)
 
-    sampler = sampler_cls(
-        continue_sampling=continue_sampling,
-        likelihoodComputationChain=chain,
-        storageUtil=HDFStorageUtil(file_prefix),
-        filePrefix=file_prefix,
-        reuseBurnin=reuse_burnin,
-        pool=mcmc_options.get(
-            "pool", ProcessPoolExecutor(max_workers=mcmc_options.get("threadCount", 1))
-        ),
-        **mcmc_options,
-    )
+    if use_multinest:
+        def likelihood(p, ndim, nparams):
+            input = [
+                params[i][1] + p[i] * (params[i][2] - params[i][1]) for i in range(ndim)
+            ]
+            return chain.computeLikelihoods(
+                chain.build_model_data(
+                    Params(*[(k, v) for k, v in zip(params.keys, input)])
+                )
+            )
 
-    # The sampler writes to file, so no need to save anything ourselves.
-    sampler.startSampling()
+        def prior(p, ndim, nparams):
+            p = [params[i][1] + p[i] * (params[i][2] - params[i][1]) for i in range(ndim)]
 
-    return sampler
+        try:
+            sampler = run(
+                likelihood,
+                prior,
+                n_dims=len(params.keys),
+                n_params=len(params.keys),
+                n_live_points=n_live_points,
+                resume=continue_sampling,
+                write_output=write_output,
+                outputfiles_basename=datadir + model_name,
+                max_iter=max_iter,
+                importance_nested_sampling=importance_nested_sampling,
+                multimodal=multimodal,
+                evidence_tolerance=evidence_tolerance,
+                sampling_efficiency=sampling_efficiency,
+                init_MPI=False,
+            )
+            return 1
+
+        except OSError:
+            raise ImportError("You also need to build MultiNest library")
```

Comment:
Let's point the user to https://johannesbuchner.github.io/PyMultiNest/install.html#id4 here so they know how to install it.

Reply:
ok